charts/stable/synapse/templates/_configmap.tpl

```diff
@@ -86,6 +86,26 @@ data:
 
     enable_registration: {{ .Values.matrix.registration.enabled }}
 
+    {{- if .Values.matrix.registration.enabled }}
+    registration_requires_token: {{ .Values.matrix.registration.requiresToken }}
+
+    {{- if .Values.matrix.registration.require3PID }}
+    registrations_require_3pid:
+        {{- range .Values.matrix.registration.requires3PID }}
+        - {{ . | .medium }}
+        {{- end }} # end range requires3PID
+    {{- end }} # end if requires3PID
```

Comment:
Those are yaml comments, and will end up in the config file, which will make zero sense to whoever will look into it later.
Use helm comments if you have to.
But I think commends on what ends where is not needed, specially on those small blocks